inc/spbc-scanner.php

```
@@ -2104,63 +2104,91 @@ function spbc_restore_file_from_backup_ajax_action_handler($id)
 
     // Success: remove all data about backup
     try {
-        $backup_deleted = unlink($full_backup_path);
-
-        if ($backup_deleted === false) {
-            return array('error' => esc_html__('Deleting backup error: Permissions denied.', 'security-malware-firewall'));
-        }
-
-        // Remove from backup
-        $sql_prepared = $wpdb->prepare(
-            'DELETE '
-            . ' FROM ' . SPBC_TBL_BACKUPED_FILES
-            . ' WHERE real_path="%s";',
-            $file_path
-        );
-        $delete = $wpdb->query($sql_prepared);
-
-        if (is_null($delete)) {
-            return array('error' => esc_html__('Error: Something is wrong during deleting backup.', 'security-malware-firewall'));
+        delete_backup_file($full_backup_path);
+        remove_backup_from_db($file_path);
+        update_cure_log($file_path);
+        if ($backup_prev_results_state !== null) {
+            update_scan_results($file_path, $backup_prev_results_state);
         }
+    } catch (Exception $e) {
+        throw new Exception($e->getMessage());
```

Are you sure it is need to throw an exception here? Maybe It will need to return array with an error?

---

Changed it in the new PR